### DISCUSSION TITLE
Cleaner way to connect with ethers5 react

### DISCUSSION THREAD

**Discussion by sbauschl** [WalletConnect/walletconnect-monorepo#3936] [Developer Support] [answered]
Looking for a simpler way to connect the wallet without hooks. (using react) Is there a way just to await the web3Modal for the provider/signer response?
Right now I have some hacky combination of useWeb3ModalEvents and useWeb3ModalSigner useEffects to monitor when the user logs in. But even that is weird because the 'CONNECT_SUCCESS' of events doesn't guarentee the provider is set yet so I have these other timing hacks in place.

What I would prefer to do is just encapsulate the login/signing process to just await getWeb3Stuff()

**Comment by sbauschl** [comment]
Opps, forgot to add. What's the easiest way to disconnect programatically? Couldn't find anything in the docs.

**Comment by glitch-txs** [comment, marked as answer]
Hi, you can use the JavaScript functions. 
https://docs.walletconnect.com/web3modal/javascript/actions?platform=ethers#subscribe-to-theme-changes

from 

```ts
const modal = createWeb3Modal({
  ethersConfig: defaultConfig({ metadata }),
  chains: [mainnet],
  projectId
})
```
